File: sci.py
artifact_weight = 5
facial_weight = 5

artifact_list = 'TFTFT'
artifact_list = list(artifact_list)
facial_list = 'FFTFT'
facial_list = list(facial_list)
correct_list = 'FTFFT'
correct_list = list(correct_list)

def weight_change(set, weight_list, w1, w2):
    for i in range(len(weight_list)):
        if weight_list[i] == set[i]:
            if w1 == 10:
                w1 += 0
                if w2 == 0:
                    continue
                else:
                    w2 -= 1
            else:
                w1 += 1
                if w2 == 0:
                    continue
                else:
                    w2 -= 1
    for i in range(len(weight_list)):
        if weight_list[i] != set[i]:
            if w1 == 0:
                w1 -= 0
                if w2 == 10:
                    continue
                else:
                    w2 += 1
            else:
                w1 -= 1
                if w2 == 10:
                    continue
                else:
                    w2 += 1
    tuple = (w1, w2)
    return tuple

tuple = weight_change(correct_list, artifact_list, artifact_weight, facial_weight)
artifact_weight = tuple[0]
facial_weight = tuple[1]

tuple = weight_change(correct_list, facial_list, facial_weight, artifact_weight)
facial_weight = tuple[0]
artifact_weight = tuple[1]


print("Artifact Weight:", artifact_weight)
print("Facial Weight:", facial_weight)

File: sci.cpp
#include <iostream>
#include <tuple>
using namespace std;

class Weight
{
  public:
    tuple<int, int> weightAsses(char correct_list[], char compare_list[], int weight_1, int weight_2);
};

tuple<int, int> Weight::weightAsses(char correct_list[], char compare_list[], int weight_1, int weight_2)
{
    for (int i = 0; i < 5; i++) {
        if (compare_list[i] == correct_list[i]) {
            if (weight_1 == 10) {
                weight_1 += 0;
                if (weight_2 == 0) {
                    continue;
                }
                else {
                    weight_2 -= 1;
                }
              }
              else {
                  weight_1 += 1;
                  if (weight_2 == 0) {
                      continue;
                  }
                  else {
                      weight_2 -= 1;
                  }
              }
        }
        else {
          if (weight_1 == 0) {
              weight_1 -= 0;
              if (weight_2 == 10) {
                  continue;
              }
              else {
                  weight_2 += 1;
              }
            }
            else {
                weight_1 -= 1;
                if (weight_2 == 0) {
                    continue;
                }
                else {
                    weight_2 += 1;
                }
            }
        }
      }

        tuple<int, int> tuple(weight_1, weight_2);
        return tuple;

  }


int main()
{
  Weight w;

  int artifact_weight = 5;
  int facial_weight = 5;

  string artifact = "FFFTTTFFFF";
  char artifact_list[artifact.length()];
  int i = 0;
  for (i = 0; i < sizeof(artifact_list); i++) {
        artifact_list[i] = artifact[i];
  }

  string facial = "TTFTTTFFTF";
  char facial_list[facial.length()];
  for (i = 0; i < sizeof(facial_list); i++) {
        facial_list[i] = facial[i];
  }

  string correct = "TTFTTTFTTF";
  char correct_list[correct.length()];
  for (i = 0; i < sizeof(correct_list); i++) {
        correct_list[i] = correct[i];
  }

  tie(artifact_weight, facial_weight) = w.weightAsses(correct_list, artifact_list, artifact_weight, facial_weight);
  tie(facial_weight, artifact_weight) = w.weightAsses(correct_list, facial_list, facial_weight, artifact_weight);

  cout << "Artifact Weight: " << artifact_weight << endl;
  cout << "Facial Weight: " << facial_weight << endl;

  return 0;
}
